core/src/main/resources/error/error-classes.json

```diff
@@ -23,7 +23,7 @@
     "message" : [ "Cannot up cast <value> from <sourceType> to <targetType>.\n<details>" ]
   },
   "CAST_INVALID_INPUT" : {
-    "message" : [ "The value <value> of the type <sourceType> cannot be cast to <targetType> because it is malformed. To return NULL instead, use `try_cast`. If necessary set <config> to \"false\" to bypass this error." ],
+    "message" : [ "The value <value> of the type <sourceType> cannot be cast to <targetType> because it is malformed. Correct the syntax for the value before casting it, or change the type to one appropriate for the value. To return NULL instead, use `try_cast`. If necessary set <config> to \"false\" to bypass this error." ],
```

Comment:
What do you mean by "Correct the syntax for the value"? Do you mean "Correct the values as per the syntax"?

Reply:
> Do you mean "Correct the values as per the syntax"?

Yep

> nit: "change the type to one appropriate for the value" => "change an appropriate type for the value"?

Maybe just, "change the target type"?

How about:
`Correct the value as per the syntax, or change its target type.`?